modules/es-extensions/publisher/modules/greg-publisher-api.js

```diff
@@ -468,5 +469,41 @@ var gregAPI = {};
             discoveryEnumData.orphanArtifactStrategy.push(OrphanArtifactStrategy.values()[index].name());
         }
         return discoveryEnumData;
+    };
+
+    gregAPI.password.addNewPassword = function (session, type, key, value) {
+        var am = assetManager(session, type);
+        var registry = am.registry.registry;
+        // Collection path used to store key and encrypted password value.
+        var path = "/_system/config/repository/components/secure-vault";
+        var resource;
+
+        if(registry.resourceExists(path)){
+            resource = registry.get(path);
+        }
+        else {
+            resource = registry.newCollection();
+        }
+
+        // Osgi service used to encrypt password.
+        var securityService =  carbon.server.osgiService('org.wso2.carbon.registry.security.vault.service.RegistrySecurityService');
+        var properties = [];
+        properties[1] = "";
+        if (key != null && value != null){
+            var encryptedText = securityService.doEncrypt(value);
+            resource.setProperty(key, encryptedText);
+            registry.beginTransaction();
+            registry.put(path, resource);
+            registry.commitTransaction();
```

Comment:
why we need to handle the transaction here. IMO, this is already handled in registry level.